CalmspaceUITests/CalmspaceUITests.swift

```diff
@@ -18,38 +18,45 @@ final class CalmspaceUITests: XCTestCase {
         app.launch()
     }
     
-    func testFullNavigationFlow() throws {
+    private func navigateToSuggestionView() {
         XCTAssertTrue(app.buttons["suggestActivity"].waitForExistence(timeout: 10))
         app.buttons["suggestActivity"].tap()
-        
-        var tapCount = 0
-        
-        while tapCount < 4 {
+    }
+ 
+    private func selectAnswers() {
+        for _ in 0..<4 {
             XCTAssertTrue(app.buttons["answerButton1"].waitForExistence(timeout: 10))
             app.buttons["answerButton1"].tap()
-            tapCount += 1
         }
-        
+    }
+    
+    private func navigateToActivitySelectionView() {
         XCTAssertTrue(app.buttons["commandButton"].waitForExistence(timeout: 10))
         app.buttons["commandButton"].tap()
-        
-        XCTAssertTrue(app.buttons["meditationButton"].waitForExistence(timeout: 10) || app.buttons["yogaButton"].waitForExistence(timeout: 10))
-        
+    }
+    
+    private func selectActivity() {
         if app.buttons["meditationButton"].exists {
             app.buttons["meditationButton"].tap()
         } else if app.buttons["yogaButton"].exists {
             app.buttons["yogaButton"].tap()
         }
-        
+    }
+    
+    private func selectActivityOption() {
         if app.cells["meditation_0"].waitForExistence(timeout: 10) {
             app.cells["meditation_0"].tap()
         } else if app.cells["yoga_0"].waitForExistence(timeout: 10) {
             app.cells["yoga_0"].tap()
         }
-        
+    }
+    
+    private func navigateToDetailsView() {
```

Comment:
In a test, there are two main steps:
1. Set up the desired conditions
2. Check if the state (or output, or whatever) meets our expectations.

In practice, it is something like this:

```swift
private func testButtonAction() {
    let button = // make or get a button 
    let selection = // some selection controlled by button
    XCTAssert(selection == 0)
    button.tap()
    XCTAssert(selection == 1)
}
```

A test usually ends with some kind of assertion. A rule of thumb is to test one thing per test, so usually, it has one assertion. In an example above though we have two assertions to make sure the state before button push is the one we expect, and then it changes in a way we expect.